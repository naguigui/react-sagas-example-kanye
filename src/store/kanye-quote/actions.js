import { createAction } from "redux-actions";
import * as kanyeQuoteActionTypes from "./constants";

export const kanyeQuoteRequest = createAction(
  kanyeQuoteActionTypes.KANYE_QUOTE_REQUEST
);
export const kanyeQuoteSuccess = createAction(
  kanyeQuoteActionTypes.KANYE_QUOTE_SUCCESS
);
export const kanyeQuoteError = createAction(
  kanyeQuoteActionTypes.KANYE_QUOTE_ERROR
);
