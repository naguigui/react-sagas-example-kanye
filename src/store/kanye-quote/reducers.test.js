import kanyeQuoteReducer from "./reducers";
import * as kanyeQuoteActions from "./actions";

const initialState = {
  isError: false,
  isLoading: false,
  quote: ""
};

describe("kanyeQuoteReducer", () => {
  it("should return the initial state", () => {
    const result = kanyeQuoteReducer(initialState, {});
    expect(result).toEqual(initialState);
  });
  it("initializes initial state if not provided", () => {
    const result = kanyeQuoteReducer(undefined, {});
    expect(result).toEqual(initialState);
  });

  describe("KANYE_QUOTE_REQUEST action", () => {
    it("sets isLoading to true", () => {
      const action = kanyeQuoteActions.kanyeQuoteRequest();
      const result = kanyeQuoteReducer(initialState, action);
      expect(result.isLoading).toBe(true);
      expect(result).toMatchSnapshot("new state");
    });
  });

  describe("KANYE_QUOTE_SUCCESS action", () => {
    it("updates quote with a payload and sets isLoading and isError to false", () => {
      const payload = {
        quote: "I am real quote"
      };
      const action = kanyeQuoteActions.kanyeQuoteSuccess(payload);
      const result = kanyeQuoteReducer(initialState, action);
      expect(result.isLoading).toBe(false);
      expect(result.isError).toBe(false);
      expect(result).toMatchSnapshot("new state");
    });
  });

  describe("KANYE_QUOTE_ERROR action", () => {
    it("sets isError to true", () => {
      const action = kanyeQuoteActions.kanyeQuoteError();
      const result = kanyeQuoteReducer(initialState, action);
      expect(result.isError).toBe(true);
      expect(result).toMatchSnapshot("new state");
    });
  });
});
