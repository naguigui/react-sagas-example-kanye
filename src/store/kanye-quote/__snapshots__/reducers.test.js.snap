// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kanyeQuoteReducer GET_KANYE_QUOTE_ERROR action sets isError to true: new state 1`] = `
Object {
  "isError": true,
  "isLoading": false,
  "quote": "",
}
`;

exports[`kanyeQuoteReducer GET_KANYE_QUOTE_REQUEST action sets isLoading to true: new state 1`] = `
Object {
  "isError": false,
  "isLoading": true,
  "quote": "",
}
`;

exports[`kanyeQuoteReducer GET_KANYE_QUOTE_SUCCESS action updates quote with a payload and sets isLoading and isError to false: new state 1`] = `
Object {
  "isError": false,
  "isLoading": false,
  "quote": "I am real quote",
}
`;
